core/include/kernel/interrupt.h

```diff
@@ -265,6 +280,23 @@ TEE_Result interrupt_configure(struct itr_chip *chip, size_t itr_num,
  */
 TEE_Result interrupt_add_handler(struct itr_handler *hdl);
 
+/*
+ * interrupt_add_main_handler() - Register a main controller interrupt handler
+ * @hdl		Interrupt handler to register
+ *
+ * @hdl.chip is expected either already set to main interrupt controller
+ * or NULL in which case it is set to interrupt_main_get().
+ */
+static inline TEE_Result interrupt_add_main_handler(struct itr_handler *hdl)
```

Comment:
I'm not sure we need `interrupt_add_main_handler()` and `interrupt_alloc_add_main_handler()`. It's not too much for the caller to call `interrupt_main_get()` first.

Reply:
ok.